(DOCSP-24585): Flutter - Remove Flex Sync Subscription by Object Type (#2251)
mongodb/docs-realm: source/sdk/flutter/sync/manage-sync-session.txt

- Remove a single subscription with the given query
- Remove a single subscription with the given name
- Remove a single subscription with the subscription reference

> - Remove all subscriptions for the given object type

Reviewer: ```suggestion
- Remove all subscriptions for a Realm object type
```

   :language: dart
   :emphasize-lines: 1, 3

Remove All Subscriptions for an Object Type
```````````````````````````````````````````

> You can remove all subscriptions for a given object type. Within a subscription 

Reviewer: ```suggestion
You can remove all subscriptions for a given Realm object type. Within a subscription 
```

update block, pass the object type to

> :flutter-sdk:`MutableSubscriptionSet.removeByType() <realm/MutableSubscriptionSet/removeByType.html>`.

Reviewer: to say "pass the object type to MutableSubscriptionSet.removeByType()" isn't technically correct b/c you're using [generics](https://dart.academy/generics-in-dart-and-flutter/) to get the type, rather than "passing" the type as an argument.

instead something like the following could be more precise:

```rst
Within a subscription 
update block, include the object type in 
:flutter-sdk:`MutableSubscriptionSet.removeByType<T extends RealmObject>() <realm/MutableSubscriptionSet/removeByType.html>`.
```

or more vague but still correct (similar to what we do when documenting other methods that use generics):

```rst
Within a subscription 
update block, call 
:flutter-sdk:`MutableSubscriptionSet.removeByType() <realm/MutableSubscriptionSet/removeByType.html>`.
```

Author: Ahh, good context - thanks! I think I'll let the code example/API method signature speak to the specifics and use fewer words. Going with your second suggestion here.

Reviewer: yeah it's hard to elegantly say what's going on there. when writing the first suggestion above, i was looking for language we could borrow and couldn't find any concise examples to borrow from.
